Source/Orts.Simulation/Common/Scripting/TrainControlSystem.cs

```diff
@@ -692,16 +701,20 @@ public enum MonitoringStatus
     public struct SignalFeatures
     {
         public readonly string MainHeadSignalTypeName;
+        public readonly string SignalTypeName;
         public readonly Aspect Aspect;
+        public readonly string DrawStateName;
         public readonly float DistanceM;
         public readonly float SpeedLimitMpS;
         public readonly float AltitudeM;
         public readonly string TextAspect;
 
-        public SignalFeatures(string mainHeadSignalTypeName, Aspect aspect, float distanceM, float speedLimitMpS, float altitudeM, string textAspect = "")
+        public SignalFeatures(string mainHeadSignalTypeName, string signalTypeName, Aspect aspect, string drawStateName, float distanceM, float speedLimitMpS, float altitudeM, string textAspect = "")
```

Comment:
Strings are objects passed by value by default. but I am not sure passing them by reference here makes sense.
Anyway, using the string type here is necessary for the information we want to transmit.

Reply:
@pzgulyas Strings are passed by reference but are immutable. You are creating a new string object each time you want to modify a string.
Since we don't modify the passed string here, we are not creating a new string, but a new reference to that same string.
So there is no problem here.